pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1895,29 +2022,21 @@ impl<'a> DatadirModification<'a> {
     }
 
     fn put_data(&mut self, key: CompactKey, val: Value) {
-        let val_serialized_size = val.serialized_size().unwrap() as usize;
-
-        // If this page was previously zero'd in the same WalRecord, then drop the previous zero page write.  This
-        // is an optimization that avoids persisting both the zero page generated by us (e.g. during a relation extend),
-        // and the subsequent postgres-originating write
-        if self.pending_zero_data_pages.remove(&key) {
-            self.pending_bytes -= ZERO_PAGE.len();
-        }
-
-        self.pending_bytes += val_serialized_size;
-        self.pending_data_pages
-            .push((key, self.lsn, val_serialized_size, val))
+        let batch = self
+            .pending_data_batch
+            .get_or_insert(SerializedValueBatch::default());
```

Comment:
I'm surprised that clippy doesn't give a warning there as each call would create a temporary SerializedValueBatch struct...

```
.get_or_insert_with(SerializedValueBatch::default);
```

Reply:
Thanks. Updated in https://github.com/neondatabase/neon/pull/9579/commits/95df95a81ae8caa819cb3bd2ba95367a27c64f2f. None of these were on the hot path, but still.